linux/src/file_linux.c

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "file.h"
+#include "file_internal.h"
+
+MU_DEFINE_ENUM_STRINGS(FILE_ASYNC_OPERATION, FILE_ASYNC_OPERATION_VALUES)
+
+typedef struct FILE_HANDLE_DATA_TAG
+{
+    int handle;
+    FILE_REPORT_FAULT user_report_fault_callback;
+    void* user_report_fault_context;
+}FILE_HANDLE_DATA;
+
+typedef struct FILE_LINUX_WRITE_TAG
+{
+    FILE_HANDLE_DATA handle;
+    FILE_WRITE_DATA_CONTEXT write_data;
+}FILE_LINUX_WRITE;
+
+typedef struct FILE_LINUX_READ_TAG
+{
+    FILE_HANDLE_DATA handle;
+    FILE_READ_DATA_CONTEXT read_data;
+}FILE_LINUX_READ;
+
+static void on_file_write_complete_linux( FILE_LINUX_WRITE* write_info);
+static void on_file_read_complete_linux( FILE_LINUX_READ* read_info);
```

Comment:
>; [](start = 68, length = 1)

file needs to end with newline

Reply:
Done in next iteration.

---
In reply to: [452471582](https://github.com/Azure/azure-c-pal/pull/7#discussion_r452471582) [](ancestors = 452471582)